### PR TITLE
feat(Unstable_IconButton): initial implementation

### PR DESCRIPTION
**Pull request by WilliamKelley**
Part of #350 

Added a new decorator in an effort to have Chromatic fully capture the focus indicator -- [margin didn't work in the previous attempt for "Unstable_Button"](https://www.chromatic.com/component?appId=60b7f55f55fbd4004993da4c&csfId=ps-unstable-button--size-by-variant-focus-visible&buildNumber=722&k=6287d0d775c890004a1ce17e-1200-snapshot-true&h=15&b=-5); I'm assuming because either their tool disregards it or there's some margin collapsing going on or something.

_Edit: [the new decorator works](https://www.chromatic.com/test?appId=60b7f55f55fbd4004993da4c&id=628d4ef1de6ce4003a0a8b64)!_

### CODE REVIEW

**Review by slingner — APPROVED** [review]

File: libs/spark/src/Unstable_IconButton/Unstable_IconButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+export const SparkThemeProvider: Story = Template.bind({});
+SparkThemeProvider.decorators = [sparkThemeProvider];
+SparkThemeProvider.storyName = '(SparkThemeProvider)';
+
+const variants: Array<Unstable_IconButtonProps['variant']> = [
+  'primary',
+  'stroked',
+  'ghost',
+];
+const sizes: Array<Unstable_IconButtonProps['size']> = ['medium', 'small'];
```
Comment (shadoath) [inline libs/spark/src/Unstable_IconButton/Unstable_IconButton.stories.tsx:55]: We don't have large?

File: libs/spark/src/Unstable_IconButton/Unstable_IconButton.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+const Unstable_IconButton: OverridableComponent<Unstable_IconButtonTypeMap> = React.forwardRef(
+  function Unstable_IconButton(props, ref) {
+    const {
+      classes: classesProp,
+      disabled,
+      size = 'medium',
+      variant = 'primary',
+      // `color` happens to be an html attribute as well; conflicts with Mui prop type
+      color: _color,
```
Comment (shadoath) [inline libs/spark/src/Unstable_IconButton/Unstable_IconButton.tsx:146]: You extract it here but don't use it. And it's not included in the type for possible values. can you explain this?

**Review by shadoath — APPROVED** [review]
YES! I'm happy to update the alert after this goes in.

File: libs/spark/src/Unstable_IconButton/Unstable_IconButton.stories.tsx (hunk 1 above)
Comment (WilliamKelley) [reply libs/spark/src/Unstable_IconButton/Unstable_IconButton.stories.tsx:55]: Nope, [figma](https://www.figma.com/file/y6Ca8BdJtQEd3Of1saRieF/components?node-id=5080%3A10852)

File: libs/spark/src/Unstable_IconButton/Unstable_IconButton.tsx (hunk 2 above)
Comment (WilliamKelley) [reply libs/spark/src/Unstable_IconButton/Unstable_IconButton.tsx:146]: I can follow the typing that variable receives, and its a non-standard attribute off of DOM typings, sepcificaly HTMLAttributes. The props inherit this off of HTMLButton (or whatever it's called) I assume.
image

**Pull request merged**